CHANGELOG.md

```diff
@@ -93,6 +93,29 @@ Non-backwards compatible changes
   and exports all the old names, but may be removed in some
   future version.
 
+### Rearrangement of algebraic Solvers
+
+* Standardised and moved the generic solver modules as follows:
+  ```agda
+  Algebra.RingSolver                        ↦ Algebra.Solver.Ring
+  Algebra.Monoid-solver                     ↦ Algebra.Solver.Monoid
+  Algebra.CommutativeMonoidSolver           ↦ Algebra.Solver.CommutativeMonoidx
+  Algebra.IdempotentCommutativeMonoidSolver ↦ Algebra.Solver.IdempotentCommutativeMonoid
+  ```
```

Comment:
Given that we went for `Data.X.Categorical` rather than `Category.X`, wouldn't it make sense to
go for `Algebra.X.Solver`? Or even `Algebra.Structures.X.Solver` given that all of these are
structures.

It would also be more consistent with the `Data.X.Solver` naming scheme.

Reply:
It's tricky, we're not very consistent between the "big three" (i.e. `Data`, `Algebra` and `Relation`). The current naming scheme is consistent within the Algebra hierarchy: `Algebra.Operations.X`, `Algebra.Properties.X` etc. 

If we were to adopt your suggestion, then we should do so for all the other `Algebra` subfolders as well... That does seem like a big change though?

Reply:
Oh I had not realised that was the case. Better leave it as it is for the moment then!

Reply:
Yup, we'll leave it for the mythical version 2.0 :laughing: